app/lib/frontend/templates/views/shared/search_banner.dart

```diff
@@ -64,3 +80,120 @@ d.Node searchBannerNode({
     ],
   );
 }
+
+/// Initial completion data is stored in an attribute.
+///
+/// This is just to the bare minimum to get started. We'll load a larger
+/// completion set from a URL when completions begin.
+final _initialCompletionData = base64.encode(utf8.encode(completionDataJson(
+  minimal: true,
+)));
+
+/// Create completion data for search input.
+///
+/// Format is dictacted by `pkg/web_app/lib/src/completion.dart`.
+///
+/// If values in `match` is a prefix of what is being typed completion
+/// will automatically start. It'll always try to use the longest match.
+/// If not specified options available are assumed to be `terminal`, that is
+/// they will be followed by whitespace.
+/// If `forcedOnly` is specified, completion can only be initiated with
+/// Ctrl+Space.
+///
+/// This can generate completion data of different sizes given [topics] and
+/// [licenses]. If [minimal] is specified, then only the most important
+/// completion keys are included.
+String completionDataJson({
```

Comment:
I'm wondering if this should be rather part of the `pkg/web_app` script we load, as it will be cached and not re-created on each page load? It could be loaded as an asynchronous library import.

Reply:
Maybe it could be two-fold, part of it may be in the `pkg/web_app`, part of it loaded dynamically from the page (like `topics`). Maybe it is not worth to optimize for this yet.

Reply:
It probably could be part of it.